src/transforms/detect_exceptions/mod.rs

```diff
@@ -0,0 +1,382 @@
+
+    /// An ordered list of fields by which to group events.
+    ///
+    /// Each group with matching values for the specified keys is reduced independently, allowing
+    /// you to keep independent event streams separate. When no fields are specified, all events
+    /// will be combined in a single group.
+    ///
+    /// For example, if `group_by = ["host", "region"]`, then all incoming events that have the same
+    /// host and region will be grouped together before being reduced.
+    #[serde(default)]
+    pub group_by: Vec<String>,
+
+    /// The interval of flushing the buffer for multiline exceptions.
+    #[serde(default = "default_multiline_flush_interval_ms")]
+    #[serde_as(as = "serde_with::DurationMilliSeconds<u64>")]
+    pub multiline_flush_interval_ms: Duration,
+
+    /// Maximum number of bytes to flush (0 means no limit). Default: 0.
+    #[serde(default = "default_max_bytes_size")]
+    pub max_bytes: usize,
+
+    /// Maximum number of lines to flush (0 means no limit). Default: 1000.
+    #[serde(default = "default_max_lines_num")]
+    pub max_lines: usize,
+}
+
+impl Default for DetectExceptionsConfig {
+    fn default() -> Self {
+        Self {
+            languages: vec![ProgrammingLanguages::All],
+            expire_after_ms: default_expire_after_ms(),
+            flush_period_ms: default_flush_period_ms(),
+            multiline_flush_interval_ms: default_multiline_flush_interval_ms(),
+            max_bytes: default_max_bytes_size(),
+            max_lines: default_max_lines_num(),
+            group_by: vec![],
+        }
+    }
+}
+
+fn default_programming_languages() -> Vec<ProgrammingLanguages> {
+    Vec::from([ProgrammingLanguages::All])
+}
+
+const fn default_expire_after_ms() -> Duration {
+    Duration::from_millis(30000)
+}
+
+const fn default_flush_period_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_multiline_flush_interval_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_max_bytes_size() -> usize {
+    0
+}
+
+const fn default_max_lines_num() -> usize {
+    1000
+}
+
+impl_generate_config_from_default!(DetectExceptionsConfig);
+inventory::submit! {
+    TransformDescription::new::<DetectExceptionsConfig>("detect_exceptions")
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "detect_exceptions")]
+impl TransformConfig for DetectExceptionsConfig {
+    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+        DetectExceptions::new(self).map(Transform::event_task)
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn outputs(&self, _: &schema::Definition) -> Vec<Output> {
+        vec![Output::default(DataType::Log)]
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "detect_exceptions"
+    }
+}
+
+pub struct DetectExceptions {
+    accumulators: HashMap<Discriminant, TraceAccumulator>,
+    languages: Vec<ProgrammingLanguages>,
```

Comment:
I'm wondering is `languages` and `group_by` can be references to the Vec's in config. They don't change once we've picked up them from configuration, we can just pass around references to them and avoid cloning.

Reply:
these get copied only once. I tried making these references, but managing lifetimes was a bit complex. If really needed, we can do it as a separate PR